src/gromacs/nblib/interactions.cpp

```diff
@@ -155,10 +158,16 @@ NonBondedInteractionMap ParticleTypesInteractions::generateTable(CombinationRule
         nonbondedParameters_[particleTypeTuple.first] = std::make_tuple(c6_combo, c12_combo);
     }
 
+    std::set<ParticleTypeName> particleTypes;
+    for (auto const& typeKey : nonbondedParameters_)
+    { // we don't need to get<1> because the list is guaranteed to be symmetric
```

Comment:
Instead of a comment this should probably be an assert.

Reply:
The symmetricity is a code invariant in this case, as can be easily proved by induction.
These types of checks are best covered by unit tests instead of general runtime assertions.
The unit test (NonBondedInteractionsAreCorrect) will catch somebody breaking the invariant in a future change.